### ISSUE TITLE
PipeChainStart with put_in/update_in etc

### ISSUE THREAD

**Issue by adamkittelson** [rrrene/credo#143]
I wanted start a discussion on an aspect of the PipeChainStart rule that I'm not sure I agree with.

The rule raises a `Pipe chain should start with a raw value.` warning on things like:

```elixir
put_in(users["john"][:age], 28)
|> ....
|> ....
```

I think that refactoring that to

```elixir
users["john"][:age]
|> put_in(28)
|> ....
|> ....
```

is somewhat confusing in that at a glance it looks like we're looking up the value returned by `users["john"][:age]` and piping it to put_in which isn't what's happening. (It still works in practice, I just think that when reading the code this way it leads to moments of confusion.)

I think it still makes sense for credo to complain about this though:
```elixir
put_in(users, ["john", :age], 28)
|> ....
|> ....
```

because in that case I think
```elixir
users
|> put_in(["john", :age], 28)
|> ....
|> ...
```
is better. I generally prefer to use this syntax anyway but sometimes I'm not able to, e.g. when working with structs.

Any thoughts?

**Comment by rrrene** [comment]
This will be fixed in the next version (see commit reference above).

**Comment by adamkittelson** [comment]
Perfect, thanks!

**Comment by rrrene** [comment]
Credo `v0.4.8` was just released and contains the fix.
